### PR TITLE
Move the tests from `/web` to the sub-test pattern

### PR DESCRIPTION
**Pull request by Peltoche**
This migration is an application of [the ADR 0002](https://github.com/cozy/cozy-stack/blob/master/docs/architecture/decisions/0002-refactor-tests-using-the-subtest-pattern.md).

It's advised to review the PR commit by commit and skipping the commit `Run the script migrate-test-init.sh on /web`. This commit is really big and have been completly generated by a script that you can review inside `./scripts`.

### CODE REVIEW

File: scripts/migrate-test-init.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+          # select the next line
+          COUNTER=$((COUNTER+1))
+          line=$(sed "${COUNTER}q;d" "$arg")
+    fi
+  done
+}
+
+
+mkdir -p /tmp/migration
```
Comment (nono) [inline scripts/migrate-test-init.sh:43]: [nitpick] You could have used `mktemp -d`

File: web/sharings/sharings_test.go [hunk 2]
```diff
@@ -117,8 +117,8 @@ func TestSharings(t *testing.T) {
 	tsB.Config.Handler.(*echo.Echo).HTTPErrorHandler = errors.ErrorHandler
 
 	// Prepare another instance for the replicator tests
```
Comment (nono) [inline web/sharings/sharings_test.go:119]: I think we no longer need this instance in this file (only used in `replicator_test.go`)

File: web/sharings/replicator_test.go [hunk 3]
```diff
@@ -39,6 +51,76 @@ func TestReplicator(t *testing.T) {
...
+	tsA.Config.Handler.(*echo.Echo).HTTPErrorHandler = errors.ErrorHandler
+
+	// Prepare Bob's browser
+	jar := setup.GetCookieJar()
+	bobUA = &http.Client{
+		CheckRedirect: noRedirect,
+		Jar:           jar,
+	}
+
+	// Prepare Bob's instance
```
Comment (nono) [inline web/sharings/replicator_test.go:86]: I don't think we need Bob's instance in replicator test

**Review by nono — APPROVED** [review]
Very nice work! Thank you

**Comment by Peltoche** [comment]
That's a fun little project :smiley: . The `/model` PR is on its way.

**Pull request merged**